src/types/decimal.rs

```diff
@@ -167,6 +168,63 @@ impl Decimal {
         scaled_coefficient *= BigUint::from(10u64).pow(exponent_delta as u32);
         UInt::BigUInt(scaled_coefficient).cmp(d2.coefficient.magnitude())
     }
+
+    /// Extract the integer and fractional parts of this as a pair of [`num_bigint::BigInt`]s
+    ///
+    /// ```ignore
+    /// # use num_bigint::BigInt;
+    /// # use ion_rs::Decimal;
+    /// let d1: Decimal = Decimal::new(123456789, -2);
+    /// let (int_part, frac_part) = d1.into_parts();
+    /// assert_eq!(int_part, BigInt::from(1234567u64));
+    /// assert_eq!(frac_part, BigInt::from(89));
+    /// ```
+    pub(crate) fn into_parts(self) -> (BigInt, BigInt) {
+        let magnitude: BigInt = self.coefficient.try_into().unwrap();
+        if self.exponent.is_zero() {
+            (magnitude, BigInt::zero())
+        } else if self.exponent.is_negative() {
+            let divisor = BigInt::from(10u64).pow((-self.exponent) as u32);
+            magnitude.div_rem(&divisor)
+        } else {
+            let multiplicand = BigInt::from(10u64).pow(self.exponent as u32);
+            (magnitude * multiplicand, BigInt::zero())
+        }
+    }
+
+    /// Returns ([`num_bigint::BigInt`], [`f64`]) representing the differences
```

Comment:
I think this was previously `(BigInt, BigInt)` -- why the switch to `f64`?

Reply:
It was.

Because, for example, with a `BigInt`, the fractional part of `1.2` was `2` and the fractional part of `1.000000002` was `2`, which does not allow the comparison that was useful.